AAUnlimited/Functions/CharInstData.cpp

```diff
@@ -67,12 +67,23 @@ void CharInstData::SetHeadTracking(int headtracking)
 void CharInstData::AddShadows(DWORD* HairPTR)
 {
 	if (this->IsValid()) {
-		if (this->m_char->m_xxSkeleton) { 
+		if (this->m_char->m_xxSkeleton) {
+			//an IsValid() check would be nice to have, but that fucks with the display tab in maker 
```

Comment:
there is an isValid check above this or you're talking about something else?
There's no situation when `CharInstData` is valid and CharacterStruct isn't.

Reply:
I believe it was kant's suggestion to add the skeleton check.

Reply:
skeleton check is fine because if it's not there it means the character didn't load its models
anyway these two ifs look adequate